2.16.0beta4 `ConcurrentModificationExceptions`

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:920)
Caused by: java.util.ConcurrentModificationException
	at java.util.LinkedList$ListItr.checkForComodification(LinkedList.java:966)
	at java.util.LinkedList$ListItr.next(LinkedList.java:888)
	at j$.util.Collection$-CC.$default$removeIf(Unknown Source:27)
	at j$.util.Collection$-EL.removeIf(Unknown Source:11)
	at com.ichi2.libanki.sched.CardQueue.remove(CardQueue.kt:13)
	at com.ichi2.libanki.sched.SchedV2.setCurrentCard(SchedV2.kt:92)
	at com.ichi2.libanki.sched.SchedV2.getCard(SchedV2.kt:49)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard$Companion.getCard(SchedulerService.kt:24)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard.execute(SchedulerService.kt:2)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard.execute(SchedulerService.kt:1)
	at com.ichi2.anki.servicelayer.AnkiTask.execute(AsyncLayer.kt:8)
	at com.ichi2.anki.servicelayer.AsyncLayerAdapterKt$asDelegate$1.task(AsyncLayerAdapter.kt:18)
	at com.ichi2.async.TaskDelegate.execTask(TaskDelegate.kt:11)
	at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.kt:229)
	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.kt:2)
	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.kt:1)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more
```
https://ankidroid.org/acra/app/1/bug/56734/report/14edf93f-ed84-406a-8c36-706be753c3dd

Going to include https://ankidroid.org/acra/app/1/bug/52580/report/524d1cff-c798-4102-afb5-15750e03064e here since it is caused by the same method

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
	at java.util.concurrent.FutureTask.run(FutureTask.java:269)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.util.NoSuchElementException
	at java.util.LinkedList.removeFirst(LinkedList.java:274)
	at java.util.LinkedList.remove(LinkedList.java:689)
	at com.ichi2.libanki.sched.CardQueue.removeFirstCard(CardQueue.kt:3)
	at com.ichi2.libanki.sched.SchedV2._getRevCard(SchedV2.kt:9)
	at com.ichi2.libanki.sched.SchedV2._getCard(SchedV2.kt:50)
	at com.ichi2.libanki.sched.SchedV2.getCard(SchedV2.kt:12)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard$Companion.getCard(SchedulerService.kt:24)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard.execute(SchedulerService.kt:2)
	at com.ichi2.anki.servicelayer.SchedulerService$GetCard.execute(SchedulerService.kt:1)
	at com.ichi2.anki.servicelayer.AnkiTask.execute(AsyncLayer.kt:8)
	at com.ichi2.anki.servicelayer.AsyncLayerAdapterKt$asDelegate$1.task(AsyncLayerAdapter.kt:18)
	at com.ichi2.async.TaskDelegate.execTask(TaskDelegate.kt:11)
	at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.kt:229)
	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.kt:2)
	at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.kt:1)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	... 4 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/CardQueue.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/CardQueue.kt
@@ -40,8 +40,9 @@ abstract class CardQueue<T : Card.Cache>(
         return queue.remove()!!.card
     }
 
-    fun remove(cid: CardId) =
-        queue.removeIf { card -> card.id == cid }
+    fun remove(cid: CardId): Boolean {
+        return queue.remove(Card.Cache(col, cid))
+    }
 
     fun add(elt: T) {
         queue.add(elt)

